src/inputs/button/__test__/button.test.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * To run this test from the church-management/client folder, execute the following command:
+ * npx jest ./src/inputs/button/__test__/button.test.js
+ */
+
+import React from 'react';
+import Button from '../button';
+import mountWithTheme from '../../../testUtils/enzymeHelpers';
+
+describe('<Button />', () => {
+    const bemName = 'some_block--some_element_name';
+    const text = 'hello world';
+
+    const props = {
+        className: bemName,
+        id: bemName,
+    };
+
+    it('should render without problems', () => {
+        const wrapper = mountWithTheme(
+            <Button
+                {...props}
+            >
+                {text}
+            </Button>,
+        );
+
+        expect(wrapper.exists()).toBe(true);
+    });
+
+    it('Should have expected \'children\'', () => {
+        const wrapper = mountWithTheme(
+            <Button
+                {...props}
+            >
+                {text}
+            </Button>,
+        );
+
+        expect(wrapper.find('span').text()).toEqual(text);
+    });
+
+    it('Should render with the root \'classes override\'', () => {
+        const rootOverride = 'makeStyles-root-123';
+
+        const wrapper = mountWithTheme(
+            <Button
+                {...props}
+                classes={{
+                    root: rootOverride,
+                }}
+            >
+                {text}
+            </Button>,
+        );
+
+        const rootNode = wrapper.find('button').first();
+
+        expect(rootNode.hasClass(rootOverride)).toEqual(true);
+    });
+
+    it('Should render with proper classNames', () => {
+        const wrapper = mountWithTheme(
+            <Button
+                {...props}
+            >
+                {text}
+            </Button>,
+        );
+
+        const rootNode = wrapper.find('button').first();
+
+        expect(rootNode.hasClass('cmui')).toEqual(true);
+        expect(rootNode.hasClass('button')).toEqual(true);
+        expect(rootNode.hasClass(/(Button)-(root)-(\d+)/)).toEqual(true);
+        expect(rootNode.hasClass(props.className)).toEqual(true);
+    });
+
+    it('Should have expected \'id\' prop', () => {
+        const wrapper = mountWithTheme(
+            <Button
+                {...props}
+            >
+                {text}
+            </Button>,
+        );
+
+        expect(wrapper.prop('id')).toEqual(props.id);
+    });
+
+    it('Should be \'disable\'', () => {
```

Comment:
Ooh the grammar hurts here! ;-)
How about "Should add expected class when \'disable\' prop is true" ?

Reply:
https://github.com/saddlebackdev/react-cm-ui/pull/325/commits/e583e8361bfabd86437159d2720386b9991b0e4f#diff-84e959d96accf1625e2aed89ccb73db78e5e20c9352a392efd06a4edaf09f3ccL91-R91

_MUCH_ better ... thank you!
Resolving.